scripts/builtin/pageRank.dml

```diff
@@ -0,0 +1,48 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+#-------------------------------------------------------------
+
+# DML builtin method for PageRank algorithm
+#
+# INPUT:
+# ------------------------------------------------------------------------------
+# G           Input Matrix
+# p           Input Vector
+# e           Input Vector
+# u           Input Vector
```

Comment:
use the following documentation

p initial page rank vector (number of nodes) - if unspecified initialized by constant value
e -> do not expose this parameter - e is an all-ones vector, generated inside the function (before the loop)
ut personalization vector (number of nodes)